sdk/core/Azure.Core/src/Serialization/ModelJsonConverter.cs

```diff
@@ -0,0 +1,82 @@
+﻿// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+using System;
+using System.Text.Json;
+using System.Text.Json.Serialization;
+
+namespace Azure.Core.Serialization
+{
+    /// <summary>
+    /// Add
+    /// </summary>
+#pragma warning disable AZC0014 // Avoid using banned types in public API
+    public class ModelJsonConverter : JsonConverter<IModelJsonSerializable<object>>
```

Comment:
Can we add a non-generic `IModelJsonSerializable` interface? Do we actually need `T` in this interface?

Reply:
do you mean do we need the `<object>`?  Yes because we don't have an IModelSerializable.